internal/transform/scope-css_test.go

```diff
@@ -0,0 +1,184 @@
+			source: "button:focus::before{}",
+			want:   "button.astro-XXXXXX:focus::before{}",
+		},
+		{
+			name:   "global children",
+			source: ".class :global(ul li){}",
+			want:   ".class.astro-XXXXXX ul li{}",
+		},
+		{
+			name:   "global universal",
+			source: ".class :global(*){}",
+			want:   ".class.astro-XXXXXX *{}",
+		},
+		{
+			name:   "global with scoped children",
+			source: ":global(section) .class{}",
+			want:   "section .class.astro-XXXXXX{}",
+		},
+		{
+			name:   "subsequent siblings + global",
+			source: ".class~:global(a){}",
+			want:   ".class.astro-XXXXXX~a{}",
+		},
+		{
+			name:   "global nested parens",
+			source: ".class :global(.nav:not(.is-active)){}",
+			want:   ".class.astro-XXXXXX .nav:not(.is-active){}",
+		},
+		{
+			name:   "global nested parens + chained class",
+			source: ":global(body:not(.is-light)).is-dark,:global(body:not(.is-dark)).is-light{}",
+			want:   "body:not(.is-light).is-dark,body:not(.is-dark).is-light{}",
+		},
+		{
+			name:   "global chaining global",
+			source: ":global(.foo):global(.bar){}",
+			want:   ".foo.bar{}",
+		},
+		{
+			name:   "class chained global",
+			source: ".class:global(.bar){}",
+			want:   ".class.astro-XXXXXX.bar{}", // technically this may be incorrect, but would require a lookahead to fix
+		},
+		{
+			name:   "chained :not()",
+			source: ".class:not(.is-active):not(.is-disabled){}",
+			want:   ".class.astro-XXXXXX:not(.is-active):not(.is-disabled){}",
+		},
+		{
+			name:   "weird chaining",
+			source: ":hover.a:focus{}", // yes this is valid. yes I’m just upset as you are :(
+			want:   ":hover.a.astro-XXXXXX:focus{}",
+		},
+		{
+			name:   "more weird chaining",
+			source: ":not(.is-disabled).a{}",
+			want:   ":not(.is-disabled).a.astro-XXXXXX{}",
+		},
+		{
+			name:   "body",
+			source: "body h1{}",
+			want:   "body h1.astro-XXXXXX{}",
+		},
+		{
+			name:   "html and body",
+			source: "html,body{}",
+			want:   "html,body{}",
+		},
+		{
+			name:   "escaped characters",
+			source: ".class\\:class:focus{}",
+			want:   ".class\\:class.astro-XXXXXX:focus{}",
+		},
+		// the following tests assert we leave valid CSS alone
+		{
+			name:   "keyframes",
+			source: "@keyframes shuffle{from{transform:rotate(0deg);}to{transform:rotate(360deg);}}",
+			want:   "@keyframes shuffle{from{transform:rotate(0deg);}to{transform:rotate(360deg);}}",
+		},
+		{
+			name:   "keyframes 2",
+			source: "@keyframes shuffle{0%{transform:rotate(0deg);color:blue;}100%{transform:rotate(360deg};}}",
+			want:   "@keyframes shuffle{0%{transform:rotate(0deg);color:blue;}100%{transform:rotate(360deg};}}",
+		},
+		{
+			name:   "charset",
+			source: "@charset \"utf-8\";",
+			want:   "@charset \"utf-8\";",
+		},
+		{
+			name:   "import (plain)",
+			source: "@import \"./my-file.css\";",
+			want:   "@import \"./my-file.css\";",
+		},
+		{
+			name:   "import (url)",
+			source: "@import url(\"./my-file.css\");",
+			want:   "@import url(\"./my-file.css\");",
```

Comment:
All tests were copied over from Astro! 🎉 And then some new ones added (to ensure we aren’t transforming CSS unintentionally)